src/features/projectsV2/ProjectsContext.tsx

```diff
@@ -188,15 +189,12 @@ export const ProjectsProvider: FC<ProjectsProviderProps> = ({
 
   useEffect(() => {
     async function loadProjects() {
-      if (page !== 'project-list' || !currencyCode) {
-        return;
-      }
-      if (projects !== null) {
-        return;
-      }
+      setProjectLocale(locale);
```

Comment:
Setting `projectLocale` here could lead to a race conditions where projectLocale is updated, but projects fail to load. Move this to after projects are successfully fetched.